Mathlib/AlgebraicGeometry/AffineSpace.lean

```diff
@@ -0,0 +1,362 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Algebra.MvPolynomial.Monad
+import Mathlib.AlgebraicGeometry.Limits
+
+/-!
+# Affine space
+
+## Main definitions
+
+- `AlgebraicGeometry.AffineSpace`: `𝔸(n; S)` is the affine `n`-space over `S`.
+- `AlgebraicGeometry.AffineSpace.coord`: The standard coordinate functions on the affine space.
+- `AlgebraicGeometry.AffineSpace.homOfVector`:
+  The morphism `X ⟶ 𝔸(n; S)` given by a `X ⟶ S` and a choice of `n`-coordinate functions.
+- `AlgebraicGeometry.AffineSpace.homOverEquiv`:
+  `S`-morphisms into `Spec 𝔸(n; S)` are equivalent to the choice of `n` global sections.
+- `AlgebraicGeometry.AffineSpace.SpecIso`: `𝔸(n; Spec R) ≅ Spec R[n]`
+
+-/
+
+open CategoryTheory Limits MvPolynomial
+
+noncomputable section
+
+namespace AlgebraicGeometry
+
+universe v u
+
+variable (n : Type v) (S : Scheme.{max u v})
+
+local notation3 "ℤ[" n "]" => CommRingCat.of (MvPolynomial n (ULift ℤ))
+local notation3 "ℤ[" n "].{" u "}" => CommRingCat.of (MvPolynomial n (ULift.{u} ℤ))
+
+/-- `𝔸(n; S)` is the affine `n`-space over `S`. -/
+def AffineSpace (n : Type v) (S : Scheme.{max u v}) : Scheme.{max u v} :=
+  pullback (terminal.from S) (terminal.from (Spec ℤ[n]))
+
+namespace AffineSpace
+
+/-- `𝔸(n; S)` is the affine `n`-space over `S`. -/
+scoped [AlgebraicGeometry] notation "𝔸("n"; "S")" => AffineSpace n S
+
+variable {n} in
+lemma of_mvPolynomial_int_ext {R} {f g : ℤ[n] ⟶ R} (h : ∀ i, f (.X i) = g (.X i)) : f = g := by
```

Comment:
Can this be moved to a file about mv polynomials?

Reply:
Mmm, maybe not because it is about the category. Not very important.

Reply:
Yeah this is quite specific about `CommRingCat.of (MvPolynomial n (ULift.{u} ℤ))` and I doubt it will be used elsewhere.